activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb

```diff
@@ -58,6 +58,14 @@ def custom_primary_key?
         options[:primary_key] != default_primary_key
       end
 
+      def valid?
+        options[:valid] == true
+      end
+
+      def novalidate?
+        options[:valid] == false
```

Comment:
If an adapter doesn't support novalidate constraint, these `valid?` and `novalidate?` returns incorrect result.
How about it?

```diff
diff --git a/activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb b/activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb
index 4332318a8e..5127d6bbe9 100644
--- a/activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb
+++ b/activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb
@@ -58,12 +58,8 @@ def custom_primary_key?
         options[:primary_key] != default_primary_key
       end
 
-      def valid?
-        options[:valid] == true
-      end
-
-      def novalidate?
-        options[:valid] == false
+      def validate?
+        options.fetch(:validate, true)
       end
 
       def defined_for?(to_table_ord = nil, to_table: nil, **options)
```